community/kernel/src/main/java/org/neo4j/kernel/impl/api/StateHandlingStatementOperations.java

```diff
@@ -653,71 +654,61 @@ public long nodeGetFromUniqueIndexSeek( KernelStatement state, NewIndexDescripto
-            Number lower, boolean includeLower,
-            Number upper, boolean includeUpper ) throws IndexNotFoundKernelException
-
-    {
-        StorageStatement storeStatement = state.getStoreStatement();
-        PrimitiveLongIterator committed = COMPARE_NUMBERS.isEmptyRange( lower, includeLower, upper, includeUpper )
-                ? PrimitiveLongCollections.emptyIterator()
-                : storeStatement.getIndexReader( index ).rangeSeekByNumberInclusive( lower, upper );
-        PrimitiveLongIterator exactMatches = filterExactRangeMatches( state, index, committed, lower, includeLower,
-                upper, includeUpper );
-        return filterIndexStateChangesForRangeSeekByNumber( state, index, lower, includeLower, upper, includeUpper,
-                exactMatches );
-    }
-
-    @Override
-    public PrimitiveLongIterator nodesGetFromIndexRangeSeekByString( KernelStatement state, NewIndexDescriptor index,
-            String lower, boolean includeLower,
-            String upper, boolean includeUpper ) throws IndexNotFoundKernelException
-
+    public PrimitiveLongIterator indexQuery( KernelStatement state, NewIndexDescriptor index,
+                                             IndexQuery...predicates ) throws IndexNotFoundKernelException
     {
         StorageStatement storeStatement = state.getStoreStatement();
         IndexReader reader = storeStatement.getIndexReader( index );
-        PrimitiveLongIterator committed = reader.rangeSeekByString( lower, includeLower, upper, includeUpper );
-        return filterIndexStateChangesForRangeSeekByString( state, index, lower, includeLower, upper, includeUpper,
-                committed );
-    }
+        PrimitiveLongIterator committed = reader.query( predicates );
 
-    @Override
-    public PrimitiveLongIterator nodesGetFromIndexRangeSeekByPrefix( KernelStatement state, NewIndexDescriptor index,
-            String prefix ) throws IndexNotFoundKernelException
-    {
-        StorageStatement storeStatement = state.getStoreStatement();
-        IndexReader reader = storeStatement.getIndexReader( index );
-        PrimitiveLongIterator committed = reader.rangeSeekByPrefix( prefix );
-        return filterIndexStateChangesForRangeSeekByPrefix( state, index, prefix, committed );
-    }
-
-    @Override
-    public PrimitiveLongIterator nodesGetFromIndexScan( KernelStatement state, NewIndexDescriptor index )
-            throws IndexNotFoundKernelException
-    {
-        StorageStatement storeStatement = state.getStoreStatement();
-        IndexReader reader = storeStatement.getIndexReader( index );
-        PrimitiveLongIterator committed = reader.scan();
-        return filterIndexStateChangesForScanOrSeek( state, index, null, committed );
+        assert predicates.length == 1: "composite indexes not yet supported";
+        IndexQuery predicate = predicates[0];
```

Comment:
Do we validate that the propertyID of the IndexQuery matches the indexdescriptor?